drm-ffi/src/mode.rs

```diff
@@ -377,28 +381,36 @@ pub fn get_connector(
         0
     };
 
-    let mut info = drm_mode_get_connector {
-        encoders_ptr: map_ptr!(&encoders),
-        modes_ptr: match modes.as_mut() {
-            Some(modes) => {
-                modes.clear();
-                modes.reserve_exact(modes_count as usize);
-                modes.as_ptr() as _
-            }
-            None => 0u64,
-        },
-        props_ptr: map_ptr!(&props),
-        prop_values_ptr: map_ptr!(&prop_values),
-        count_modes: modes_count,
-        count_props: map_len!(&props),
-        count_encoders: map_len!(&encoders),
-        connector_id,
-        ..Default::default()
-    };
+    let info = loop {
+        let mut info = drm_mode_get_connector {
+            encoders_ptr: map_ptr!(&encoders),
+            modes_ptr: match modes.as_mut() {
+                Some(modes) => {
+                    modes.clear();
+                    modes.reserve_exact(modes_count as usize);
+                    modes.as_ptr() as _
+                }
+                None => 0u64,
+            },
+            props_ptr: map_ptr!(&props),
+            prop_values_ptr: map_ptr!(&prop_values),
+            count_modes: modes_count,
+            count_props: map_len!(&props),
+            count_encoders: map_len!(&encoders),
+            connector_id,
+            ..Default::default()
+        };
 
-    unsafe {
-        ioctl::mode::get_connector(fd, &mut info)?;
-    }
+        unsafe {
+            ioctl::mode::get_connector(fd, &mut info)?;
+        }
+
+        if info.count_modes == modes_count {
```

Comment:
The test for this in [libdrm](https://gitlab.freedesktop.org/mesa/drm/-/blob/main/xf86drmMode.c) is:

```c
	if (counts.count_props < conn.count_props ||
	    counts.count_modes < conn.count_modes ||
	    counts.count_encoders < conn.count_encoders) {
```